packages/gatsby/src/services/graphql-typegen.ts

```diff
@@ -0,0 +1,60 @@
+import reporter from "gatsby-cli/lib/reporter"
+import { EventObject } from "xstate"
+import { IBuildContext } from "../internal"
+import { IDataLayerContext, IQueryRunningContext } from "../state-machines"
+import {
+  writeGraphQLFragments,
+  writeGraphQLSchema,
+} from "../utils/graphql-typegen/file-writes"
+import { writeTypeScriptTypes } from "../utils/graphql-typegen/ts-codegen"
+
+export async function graphQLTypegen(
+  {
+    program,
+    store,
+    parentSpan,
+  }:
+    | Partial<IBuildContext>
+    | Partial<IQueryRunningContext>
+    | Partial<IDataLayerContext>,
+  _: EventObject,
+  {
+    src: { compile },
+  }: {
+    src: {
+      type: string
+      compile?: "all" | "schema" | "definitions"
+    }
+  }
+): Promise<void> {
+  if (!program || !store || !compile) {
+    reporter.panic(
+      `Missing required params in graphQLTypegen. program: ${!!program}. store: ${!!store}. compile: ${!!compile}`
+    )
+  }
```

Comment:
Do we want to keep it as is (initial look says it looks like debuggy, but to be fair, this should never happen so probably we can keep it)

Reply:
I added a comment clarifying it